### DISCUSSION TITLE
/dev/ttyUSB0' port is not compatible

### DISCUSSION THREAD

**Discussion by soleluna** [jczic/ESP32-MPY-Jama#28] [Q&A]
First of all: thank you very much for this great looking GUI. 
I compiled and started ESP32-MPY-Jama  inside Linux MINT 21.
Tried to connect to microPYTHON flashed device ESP8266 with running script BLINKING-LED.
Got error message:
--- **The device on the '/dev/ttyUSB0' port is not compatible**  ---

Connection to /dev/ttyUSB0  is working with VS CODE or GTKTerm.
User is already member of group DIALOUT.
Is it only working with ESP32, not with ESP8266?

**Comment by jczic** [comment]
Hello @soleluna, thanks for your question, unfortunately I don't have an ESP8266 chipset to test this.
If you have indeed cloned the repository, you are up to date with the sources and the port connection has been completely redone so that it works very well now.
However, this does not only open the port in terminal mode but exchange with the system and also changes the REPL mode. So it is quite possible that there is an incompatibility with the ESP8266.

What version of MicroPython do you have in the firmware of your chipset?

**Reply by soleluna** [reply]
Hi Jean-Christophe,
My MPY Version on the ESP8266 is
**MicroPython v1.19.1 on 2022-06-18; ESP module (1M) with ESP8266**

Tomorrow I will get ESP32-WROOM devices.
With these I will try to connect again with your ESP32-MPY-JAMA

Is there an easy way back from MicroPython? Done with ?
**python -m esptool --chip esp32 erase_flash**

Greetings from Karlsruhe to Paris

**Reply by jczic** [reply]
Hi @soleluna, ok, the MicroPython version is good though but according to this chip, I don't know indeed.
To install MicroPython, yes, you can go through esptool directly or use the software in the "Firmware Tools" part which also uses esptool.
This uses the "bootloader" mode to connecting to the esp32 and will work.
